sdk/python/pulumi_docker/docker.py

```diff
@@ -0,0 +1,635 @@
+# Copyright 2016-2020, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import math
+import re
+import subprocess
+from random import random
+from typing import Optional, Union, List, Mapping
+from distutils.version import LooseVersion
+
+import pulumi
+from .utils import get_image_name_and_tag
+
+
+# Registry is the information required to login to a Docker registry.
+class Registry:
+    registry: pulumi.Input[str]
+    username: pulumi.Input[str]
+    password: pulumi.Input[str]
+
+    def __init__(self, registry: pulumi.Input[str], username: pulumi.Input[str], password: pulumi.Input[str]):
+        self.registry = registry
+        self.username = username
+        self.password = password
+
+
+# CacheFrom may be used to specify build stages to use for the Docker build cache. The final image
+# is always implicitly included.
```

Comment:
I know this is unchanged from the original PR - but as a note possibly for follow-up, I don't believe this style of comments will show up in IDEs or doc generators.  I believe we need to use the same PEP 257 docstring style as in (e.g) https://github.com/pulumi/pulumi-docker/blob/master/sdk/python/pulumi_docker/container.py.  (note that these really aren't comments - they are API documentation)

Reply:
Fixed to use the correct docstring style